app/assets/javascripts/discourse/controllers/move-to-topic.js.es6

```diff
@@ -0,0 +1,217 @@
+import { movePosts, mergeTopic } from "discourse/models/topic";
+import DiscourseURL from "discourse/lib/url";
+import { default as computed } from "ember-addons/ember-computed-decorators";
+import { extractError } from "discourse/lib/ajax-error";
+
+export default Ember.Controller.extend(ModalFunctionality, {
+  topicName: null,
+  saving: false,
+  categoryId: null,
+  tags: null,
+  canAddTags: Ember.computed.alias("site.can_create_tag"),
+  canTagMessages: Ember.computed.alias("site.can_tag_pms"),
+  selectedTopicId: null,
+  newTopic: Ember.computed.equal("selection", "new_topic"),
+  existingTopic: Ember.computed.equal("selection", "existing_topic"),
+  newMessage: Ember.computed.equal("selection", "new_message"),
+  existingMessage: Ember.computed.equal("selection", "existing_message"),
+  participants: null,
+
+  topicController: Ember.inject.controller("topic"),
+  selectedPostsCount: Ember.computed.alias(
+    "topicController.selectedPostsCount"
+  ),
+  selectedAllPosts: Ember.computed.alias("topicController.selectedAllPosts"),
+  selectedPosts: Ember.computed.alias("topicController.selectedPosts"),
+
+  @computed("saving", "selectedTopicId", "topicName")
+  buttonDisabled(saving, selectedTopicId, topicName) {
+    return (
+      saving || (Ember.isEmpty(selectedTopicId) && Ember.isEmpty(topicName))
+    );
+  },
+
+  @computed(
+    "saving",
+    "newTopic",
+    "existingTopic",
+    "newMessage",
+    "existingMessage"
+  )
+  buttonTitle(saving, newTopic, existingTopic, newMessage, existingMessage) {
+    if (newTopic) {
+      return I18n.t("topic.split_topic.title");
+    } else if (existingTopic) {
+      return I18n.t("topic.merge_topic.title");
+    } else if (newMessage) {
+      return I18n.t("topic.move_to_new_message.title");
+    } else if (existingMessage) {
+      return I18n.t("topic.move_to_existing_message.title");
+    } else {
+      return I18n.t("saving");
+    }
+  },
+
+  onShow() {
+    this.setProperties({
+      "modal.modalClass": "move-to-modal",
+      saving: false,
+      selection: "new_topic",
+      categoryId: null,
+      topicName: "",
+      tags: null,
+      participants: null
+    });
+
+    const isPrivateMessage = this.get("model.isPrivateMessage");
+    const canSplitTopic = this.get("canSplitTopic");
+    if (isPrivateMessage) {
+      if (canSplitTopic) {
+        this.set("selection", "new_message");
+      } else {
+        this.set("selection", "existing_message");
+      }
+    } else if (!canSplitTopic) {
+      this.set("selection", "existing_topic");
+    }
+  },
+
+  @computed("selectedAllPosts", "selectedPosts", "selectedPosts.[]")
+  canSplitTopic(selectedAllPosts, selectedPosts) {
+    return (
+      !selectedAllPosts &&
+      selectedPosts.length > 0 &&
+      selectedPosts.sort((a, b) => a.post_number - b.post_number)[0]
+        .post_type === 1
+    );
+  },
+
+  actions: {
+    performMove() {
+      if (this.get("newTopic")) {
```

Comment:
This is quite a few if statements that could be DRYer. If you had a constant above of the different modes `["newTopic", "existingTopic", etc]` you could loop through them, check if true and then send to the string name for the action. 

This could also clean up `buttonTitle` above by doing something similar.

Reply:
Good idea, done. Regarding cleaning up `buttonTitle` above I tried using array constant for that as well but since locale keys and constants we are using here have different names I couldn't use it above.